multi-tenancy-observability/README.md

```diff
@@ -0,0 +1,270 @@
+- other ideas are welcome!
+
+#### Mapping user's groups to tenants
+
+What interface could we offer our customers to enable them to define the mapping between their groups and their tenants:
+
+- Custom resources like TenantCR, GroupCR
+- UI in happa
+- other ideas ?
+
+## Write path
+
+### Current architecture for the write path
+
+On the write path, we currently use promtail to collect container and machine logs and grafana agent to collect kubernetes events.
+
+We currently need those 2 tools because:
+1. Promtail cannot get kubernetes events
+2. Grafana Agent is going through quite a big rewrite in it's configuration and all features we needed were not supported when we deployed it
+
+Here is what the flow looks like:
+1. The logging-operator creates a `password` for the cluster's logging agents to be able to send logs to Loki and configures them (actual configuration of what to scrape and so on) via an extra-config.
+2. The logging-operator adds the new cluster_id/password pair to the proxy configuration
+3. The logging agents collect and send logs to the loki-multi-tenant-proxy which validates the basic authentication sent in the http header and transform it into a tenant information (X-Scope_OrgID)
+4. Loki writes the data to storage for the configured tenant (cluster_id in that case)
+
+### Configuring multi-tenancy on the write path
+
+#### Different tenant configuration options
+
+After a few different customer calls, we found out that they have different needs when it comes to multi-tenancy.
+
+##### Cluster == Tenant
+
+This is the approach we are currently using for technical reasons. We do not think this would match with customers workload as they mostly have a lot of customers sharing clusters.
+
+##### Namespace == Tenant
+
+This approach makes it really easy for us and customers to configure tenants and it would definitely be easy to onboard teams (e.g. adding a label on the namespace would make the agents collect logs from the containers in the namespace).
+
+But this approach brings a few questions:
+1. What tenancy should common internal apps use (e.g. prometheus-operator, api-server, cilium logs are accessed by multiple internal teams)? Do we need to ensure customers are putting those apps into a namespace accessible by all?
+2. What about teams that share namespaces?
+
+##### App == Tenant
+
+This approach is especially flexible for customers but we do not think this approach makes sense because the mapping configuration would be impossible to maintain and we would probably face technical limits pretty fast
+
+##### Hybrid mode
+
+This approach would allow customers to be able to set the tenancy as they want (per namespace or per app depending on the use case).
+
+#### Implementation details
+
+The implementation details are at this stage are still an open topic, because no single tool fulfills all of our requirements. Until this RFC we use two tools, grafana-agent and promtail, which are bot not yet feature complete.
```

Comment:
```suggestion
The implementation details are at this stage still an open topic, because no single tool fulfills all of our requirements. As for the moment we wrote this RFC we used two tools, grafana-agent and promtail, which are both not yet feature complete.
```